### ISSUE TITLE
Only Audio transcoding

### ISSUE THREAD

**Issue by iNfRaSoNiC13** [jellyfin/jellyfin-tizen#69]
Hello,
can you confirm or deny that this behavior is normal? 

Jellyfin is transcoding both video and audio when only AUDIO is not supported. I know Samsung TVs * at least mine, does not support TRUEHD but why it is transcoding the video? I have the same file with DTS audio and it is directplaying.
So my question is can jellyfin transcode only audio? 
Thank you in advance. 

Video
Title4K HEVC HDR
CodecHEVC
ProfileMain 10
Level153
Resolution3840x1600
Aspect ratio2.40:1
InterlacedNo
Framerate23.976025
Bitrate32399 kbps
Video rangeHDR
Color spacebt2020nc
Color transfersmpte2084
Color primariesbt2020
Pixel formatyuv420p10le
Ref frames1
Audio
TitleEnglish - Английски - TRUEHD - 7.1 - Default
Languageeng
CodecTRUEHD
Layout7.1
Channels8 ch
Bitrate4149 kbps
Sample rate48000 Hz
Bit depth24 bit
DefaultYes
Subtitle
TitleEnglish (SDH) - Английски
Languageeng
CodecPGSSUB
DefaultNo
ForcedNo
ExternalNo
Subtitle
TitleFrench
Languagefre
CodecPGSSUB
DefaultNo
ForcedNo
ExternalNo
Subtitle
TitleSpanish - Испански
Languagespa
CodecPGSSUB
DefaultNo
ForcedNo
ExternalNo
Subtitle
TitleUndefined
CodecSRT
DefaultNo
ForcedNo
ExternalYes

**Comment by dmitrylyzo** [comment]
I can reproduce HEVC+TRUEHD -> H264+AAC transcoding.
IMO, Jellyfin should transcode only audio and copy video stream as is.
Device profile returns only `h264` for `ts` transcoding - probably that's why.

DTS is disabled for Tizen 4+: https://github.com/jellyfin/jellyfin-web/blob/f21fc934ca1fbc4ad40b7294c4e5be938d0ce2f8/src/scripts/browserDeviceProfile.js#L392-L395

**Comment by iNfRaSoNiC13** [comment]
"Device profile returns only h264 for ts transcoding - probably that's why. "
Sorry but I Can't quite get how to fix the issue or you are saying that the problem is with the video file?

**Comment by dmitrylyzo** [comment]
> Sorry but I Can't quite get how to fix the issue or you are saying that the problem is with the video file?

File is OK. Jellyfin reports wrong device profile.

You can add `hlsInTsVideoCodecs.push('hevc');` after this line https://github.com/jellyfin/jellyfin-web/blob/f21fc934ca1fbc4ad40b7294c4e5be938d0ce2f8/src/scripts/browserDeviceProfile.js#L449
and rebuild Tizen app. This is more a hack than a fix - server just grabs the first codec and will transcode all to HEVC if transcoding is required. At least, that's how I see it. "Directs" should be fine.

**Comment by iNfRaSoNiC13** [comment]
> > Sorry but I Can't quite get how to fix the issue or you are saying that the problem is with the video file?
> 
> File is OK. Jellyfin reports wrong device profile.
> 
> You can add `hlsInTsVideoCodecs.push('hevc');` after this line https://github.com/jellyfin/jellyfin-web/blob/f21fc934ca1fbc4ad40b7294c4e5be938d0ce2f8/src/scripts/browserDeviceProfile.js#L449
> and rebuild Tizen app. This is more a hack than a fix - server just grabs the first codec and will transcode all to HEVC if transcoding is required. At least, that's how I see it. "Directs" should be fine.

I will try it and report back, thank again for your work and answers.

**Comment by iNfRaSoNiC13** [comment]
> > Sorry but I Can't quite get how to fix the issue or you are saying that the problem is with the video file?
> 
> File is OK. Jellyfin reports wrong device profile.
> 
> You can add `hlsInTsVideoCodecs.push('hevc');` after this line https://github.com/jellyfin/jellyfin-web/blob/f21fc934ca1fbc4ad40b7294c4e5be938d0ce2f8/src/scripts/browserDeviceProfile.js#L449
> and rebuild Tizen app. This is more a hack than a fix - server just grabs the first codec and will transcode all to HEVC if transcoding is required. At least, that's how I see it. "Directs" should be fine.

Yup, the new line fixed the issue although now I am getting direct streaming for the audio as well or at least there is no indication of transcoding but that is not possible so I am assuming that its just displaying it at direct streaming but the audio is being transcoded.

**Comment by dmitrylyzo** [comment]
`Direct streaming` is not `Direct Play`. It means that video is not transcoded.
I think there was some description of `DirectPlay` -> `Remux` -> `DirectStream` -> `Transcode`.

In `Playback data`, you should get:
- video `HEVC (direct)` - video is simply copied unchanged
- audio `AAC` - audio is transcoded from `TRUEHD` to `AAC`

I reopen issue because as I said that was a hack not a fix.

**Comment by iNfRaSoNiC13** [comment]
> `Direct streaming` is not `Direct Play`. It means that video is not transcoded.
> I think there was some description of `DirectPlay` -> `Remux` -> `DirectStream` -> `Transcode`.
> 
> In `Playback data`, you should get:
> 
> * video `HEVC (direct)` - video is simply copied unchanged
> * audio `AAC` - audio is transcoded from `TRUEHD` to `AAC`
> 
> I reopen issue because as I said that was a hack not a fix.

Yes, and I am having issues with it that I've just noticed. Adding this line is creating a ton of small clips around 18mb each and it is flooding the HDD. So maybe there is another solution.

**Comment by dmitrylyzo** [comment]
> Yes, and I am having issues with it that I've just noticed. Adding this line is creating a ton of small clips around 18mb each and it is flooding the HDD. So maybe there is another solution.
>
> They look like this "550a418f3d77001ef136897f1d99f6ba256.ts" and are between 5 and 20mb and are being created every second. This is happening when I am watching movie with non compatible audio like DTS, TRUEHD and so on.

This is normal: if any codec is not supported, it requires transcoding (in your case `DirectStream`, because video codec is supported).
I called it a "hack" mainly because of the order dependency.

I have the same problem with some files (DTS) and thought, "It would be nice to transcode audio into a separate stream and then merge them on the client". But it requires to dive into the server source.

> Will removing the block fix the issue or we should wait for the 10.8.0 release ? Thank you.

You will get the same as now - `DirectStream`. I just want to remove the order dependency.

**Comment by iNfRaSoNiC13** [comment]
> > Yes, and I am having issues with it that I've just noticed. Adding this line is creating a ton of small clips around 18mb each and it is flooding the HDD. So maybe there is another solution.
> > They look like this "550a418f3d77001ef136897f1d99f6ba256.ts" and are between 5 and 20mb and are being created every second. This is happening when I am watching movie with non compatible audio like DTS, TRUEHD and so on.
> 
> This is normal: if any codec is not supported, it requires transcoding (in your case `DirectStream`, because video codec is supported).
> I called it a "hack" mainly because of the order dependency.
> 
> I have the same problem with some files (DTS) and thought, "It would be nice to transcode audio into a separate stream and then merge them on the client". But it requires to dive into the server source.
> 
> > Will removing the block fix the issue or we should wait for the 10.8.0 release ? Thank you.
> 
> You will get the same as now - `DirectStream`. I just want to remove the order dependency.

So in other words the best thing to do right now is to wait for a fix or choose files with multiple auido one of which is AC3?
Thank you again.

**Comment by dmitrylyzo** [comment]
> So in other words the best thing to do right now is to wait for a fix or choose files with multiple auido one of which is AC3?

Yes.

As for DirectStream/Transcode, it requires about as much free space as the original file. Those ts-files will be removed when playback is stopped.

**Comment by iNfRaSoNiC13** [comment]
> > So in other words the best thing to do right now is to wait for a fix or choose files with multiple auido one of which is AC3?
> 
> Yes.
> 
> As for DirectStream/Transcode, it requires about as much free space as the original file. Those ts-files will be removed when playback is stopped.

Ok thank you, I will use your method until a proper fix is applied. Have a good day.

**Comment by uggedal** [comment]
I see no problems with unpatched jellyfin-web 10.7.1 and jellyfin-tizen HEAD. HEVC is remuxed directly and DTS audio is transcoded to AAC.

I do have `Allow video playback that requires transcoding` disabled for my user though. Maybe that's what makes it work?

**Comment by uggedal** [comment]
I belive it's working because of this: https://github.com/jellyfin/jellyfin/blob/master/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs#L2953-L2957

**Comment by iNfRaSoNiC13** [comment]
> I belive it's working because of this: https://github.com/jellyfin/jellyfin/blob/master/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs#L2953-L2957

Can you share your video file info?

**Comment by dmitrylyzo** [comment]
There are 2 problems at the moment:
- missed `hevc` for `ts` container.
- fallback stream for HEVC HDR that doesn't work on my TV: https://github.com/jellyfin/jellyfin/issues/5576#issuecomment-806178759

Just checked - disabled `Allow video playback that requires transcoding` can workaround both.

**Comment by iNfRaSoNiC13** [comment]
> There are 2 problems at the moment:
> 
> * missed `hevc` for `ts` container.
>   [jellyfin/jellyfin-web#2519](https://github.com/jellyfin/jellyfin-web/pull/2519)
> * fallback stream for HEVC HDR, which doesn't work on my TV
>   [jellyfin/jellyfin#5576 (comment)](https://github.com/jellyfin/jellyfin/issues/5576#issuecomment-806178759)
> 
> Just checked - disabled `Allow video playback that requires transcoding` can workaround both.

Where exactly is "Allow video playback that requires transcoding" ?

**Comment by dmitrylyzo** [comment]
> Where exactly is "Allow video playback that requires transcoding" ?

`Dashboard` -> `Users` -> _Your user_ -> `Media Playback` section

**Comment by iNfRaSoNiC13** [comment]
Yup, working like a charm, thank you both.

**Comment by binarydad** [comment]
> There are 2 problems at the moment:
> 
>     * missed `hevc` for `ts` container.
>       [Add HEVC in TS for Tizen and webOS jellyfin-web#2519](https://github.com/jellyfin/jellyfin-web/pull/2519)
> 
>     * fallback stream for HEVC HDR, which doesn't work on my TV
>       [Support for multiple video codecs in transcoding profile jellyfin#5576 (comment)](https://github.com/jellyfin/jellyfin/issues/5576#issuecomment-806178759)
> 
> 
> Just checked - disabled `Allow video playback that requires transcoding` can workaround both.

Thank you for this! I was having issues with both audio AND video being transcoded on my Roku devices. Disabling this allowed it to "Direct Stream". This was a big issue for me previously. Thanks!!
